crates/flipperzero/src/gui/mod.rs

```diff
@@ -1,3 +1,230 @@
 //! GUI service.
 
 pub mod canvas;
+pub mod icon;
+pub mod icon_animation;
+pub mod view;
+pub mod view_port;
+pub mod xbm;
+
+use crate::{
+    gui::{
+        canvas::CanvasView,
+        view_port::{ViewPort, ViewPortCallbacks},
+    },
+    input::InputEvent,
+};
+use core::{
+    ffi::c_char,
+    fmt::Debug,
+    ops::{Deref, DerefMut},
+};
+use flipperzero_sys::{
+    self as sys, furi::UnsafeRecord, Canvas as SysCanvas, Gui as SysGui, GuiLayer as SysGuiLayer,
+};
+
+/// System Gui wrapper.
+pub struct Gui {
+    raw: UnsafeRecord<SysGui>,
+}
+
+impl Gui {
+    /// Furi record corresponding to GUI.
+    pub const RECORD: *const c_char = sys::c_string!("gui");
+
+    pub fn new() -> Self {
+        // SAFETY: `RECORD` is a constant
+        let gui = unsafe { UnsafeRecord::open(Self::RECORD) };
+
+        Self { raw: gui }
+    }
+
+    pub fn as_raw(&self) -> *mut SysGui {
+        self.raw.as_raw()
+    }
+
+    pub fn add_view_port<VPC: ViewPortCallbacks>(
+        &mut self,
+        view_port: ViewPort<VPC>,
+        layer: GuiLayer,
+    ) -> GuiViewPort<'_, VPC> {
```

Comment:
I don't see any examples with multiple view ports. Is that a thing we should expect to eventually support?
- If so, then storing `&self` inside `GuiViewPort` currently prevents this (because then we can't re-take a mutable reference to `self`), so we might need an alternative API for adding multiple view ports at once (or an alternative way to reference the parent instead of `&self`, but that still binds to the lifetime of `Self` rather than the `&mut self` or `&self`).
- If not, then `Gui::add_view_port` feels misnamed, and it should instead be something like `Gui::connect_view_port`.